common/src/main/java/org/keycloak/common/util/KeycloakUriBuilder.java

```diff
@@ -458,13 +458,15 @@ private String buildString(Map<String, ?> paramMap, boolean fromEncodedMap, bool
             buffer.append(ssp);
         } else if (userInfo != null || host != null || port != -1) {
             buffer.append("//");
-            if (userInfo != null)
+            if (userInfo != null) {
+                if ("".equals(host)) throw new RuntimeException("empty host name, but userInfo supplied");
```

Comment:
Maybe it's better to compare also with null, something like:

```java
if (host == null || host.isEmpty()) ...
```

Reply:
While the Null-Check was already done of the upper else-if, I added that in :+1: